data_handler/handlers/helpers.py

```diff
@@ -72,6 +75,104 @@ def set_value(self, token: str, value: decimal.Decimal):
+        self.current_block = current_block
+        self.current_timestamp = last_block_data.timestamp if last_block_data else 0
+
+        self.cumulative_collateral: dict[str, Decimal] = {}
+        self.cumulative_debt: dict[str, Decimal] = {}
+        self.token_timestamps: dict[str, int] = {}
+        self._fill_state_data()
+
+    def get_seconds_passed(self, token_name: str) -> Decimal:
+        """
+        Get the number of seconds passed since the last event for given token.
+        :param token_name: str - The name of the token, for example `STRK`.
+        :return: Decimal - The number of seconds passed.
+        """
+        return Decimal(
+            self.current_timestamp - self.token_timestamps[token_name]
+        )
+
+    def update_state_cumulative_data(
+            self, token_name: str, current_block: int, collateral_change: Decimal, debt_change: Decimal
+    ) -> None:
+        """
+        Update the state of interest rate calculation with the new data.
+        :param token_name: str - The name of the token, for example `STRK`.
+        :param current_block: int - The current block number.
+        :param collateral_change: Decimal - The change in collateral(supply) interest rate.
+        :param debt_change: Decimal - The change in debt(borrow) interest rate.
+        """
+        self.cumulative_collateral[token_name] += collateral_change
+        self.cumulative_debt[token_name] += debt_change
+        self.token_timestamps[token_name] = self.current_timestamp
+        self.current_block = current_block
+
+    def _fill_state_data(self) -> None:
+        """General function for filling the state data."""
+        self._fill_cumulative_data()
+        self._fill_timestamps()
+
+    def _fill_cumulative_data(self) -> None:
+        """Fill the cumulative collateral and debt data with latest block data or default values. Default value is 1."""
+        if self.last_block_data:
+            self.cumulative_collateral = {
+                token_name: Decimal(value) for token_name, value in self.last_block_data.collateral.items()
+            }
+            self.cumulative_debt = {
+                token_name: Decimal(value) for token_name, value in self.last_block_data.debt.items()
+            }
+        else:
+            self.cumulative_collateral = {
+                token_name: Decimal("1") for token_name in TOKEN_MAPPING.values()
+            }
+            self.cumulative_debt = self.cumulative_collateral.copy()
+
+    def _fill_timestamps(self) -> None:
+        """Fill the token timestamps with latest block timestamp or default value. Default value is 0"""
+        if self.last_block_data:
+            self.token_timestamps = {
+                token_name: self.last_block_data.timestamp for token_name in TOKEN_MAPPING.values()
+            }
+        else:
+            self.token_timestamps = {
+                token_name: 0 for token_name in TOKEN_MAPPING.values()
```

Comment:
Won't this cause the first period to be very long and hence the accumulated interest to be very high?

Reply:
Added comment for clarification, code that updates timestamps is [there](https://github.com/CarmineOptions/derisk-research/pull/143/files#r1662462015).